lms/djangoapps/bookmarks/views.py

```diff
@@ -0,0 +1,301 @@
+"""
+HTTP end-points for the Bookmarks API.
+
+For more information, see:
+https://openedx.atlassian.net/wiki/display/TNL/Bookmarks+API
+"""
+import logging
+
+from django.core.exceptions import ObjectDoesNotExist, MultipleObjectsReturned
+from django.utils.translation import ugettext as _
+
+from rest_framework import status
+from rest_framework import permissions
+from rest_framework.authentication import OAuth2Authentication, SessionAuthentication
+from rest_framework.generics import ListCreateAPIView
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey, UsageKey
+
+from openedx.core.lib.api.serializers import PaginationSerializer
+
+from xmodule.modulestore.exceptions import ItemNotFoundError
+from xmodule.modulestore.django import modulestore
+
+from .api import get_bookmark
+from .models import Bookmark
+from .serializers import BookmarkSerializer
+
+
+log = logging.getLogger(__name__)
+
+DEFAULT_FIELDS = ["id", "course_id", "usage_id", "created"]
+OPTIONAL_FIELDS = ['display_name', 'path']
+
+
+class BookmarksView(ListCreateAPIView):
+    """
+    **Use Case**
+
+        Get a paginated list of bookmarks in particular course.
```

Comment:
If you don't pass course_id as a parameter, will you get all bookmarks, across all courses?